File: spring-integration/subscribable-channels/src/main/java/com/ddubson/services/UppercasePrintService.java
package com.ddubson.services;

import org.springframework.messaging.Message;

public class UppercasePrintService {
    public void print(Message<String> msg) {
        System.out.format("[%s] %s\n", Thread.currentThread().getName(), msg.getPayload().toUpperCase());
    }
}

File: spring-integration/subscribable-channels/messaging-bridge/src/main/kotlin/com/ddubson/integration/UppercasePrintService.kt
package com.ddubson.integration

import com.ddubson.ANSIColor
import com.ddubson.LogAdapter
import org.springframework.messaging.Message
import org.springframework.stereotype.Component

@Component("uppercasePrintService")
class UppercasePrintService(private val lg: LogAdapter) {
    fun print(msg: Message<String>) {
        lg.info(
                "[2.1] [${Thread.currentThread().name}] ${msg.payload.toUpperCase()}"
                , ANSIColor.ANSI_CYAN)

    }
}